roles/openshift_health_checker/openshift_checks/etcd_volume.py

```diff
@@ -0,0 +1,54 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for warning about etcd volume size past a defined threshold.
+"""
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class EtcdVolume(OpenShiftCheck):
+    """Ensure disk size for an etcd host does not exceed a defined limit"""
+
+    name = "etcd_volume"
+    tags = ["etcd"]
+
+    etcd_default_size_limit = 0
+
+    def run(self, tmp, task_vars):
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+
+        etcd_mount_path = self.get_etcd_mount_path(ansible_mounts)
+        etcd_disk_size_available = int(etcd_mount_path["size_available"])
+        etcd_disk_size_total = int(etcd_mount_path["size_total"])
+        etcd_disk_size_used = etcd_disk_size_total - etcd_disk_size_available
+
+        self.etcd_default_size_limit = 0.9 * etcd_disk_size_total
```

Comment:
The current shape of the code makes it harder to understand at a glance what the default limit is:

```python

class ...:
    etcd_default_size_limit = 0   # §1

    def run(...):
        ...
        self.etcd_default_size_limit = 0.9 * etcd_disk_size_total  # §2
```

Whoever stares at **§1** will wonder "hmm, the default limit is zero".

But, in fact, if one keeps reading through `run` until **§2**, one notices that the "default default" is unconditionally updated at runtime to be 90% of some other value.

## Proposal

Since we are making the default be 90% of the total size, and based on a previous experience doing a similar task, I believe that a perhaps better initial interface would be to set the limit in percentage, instead of absolute values.

Such that we could write:

```python
class ...:
    default_etcd_size_limit_percent = 0.9

    def run(...):
        ...
        etcd_size_limit_percent = float(get_var(task_vars, "etcd_size_limit_percent", default=self.default_etcd_size_limit_percent))

        if etcd_disk_size_used / etcd_disk_size_total > etcd_size_limit_percent:
            ...
```

This would make it easier for readers to see what the default is, and for users to specify values such as "0.85" (or "85"?), instead of number of bytes like "34000000".